src/components/kpiWidget/barChart.js

```diff
@@ -91,7 +91,7 @@ const sortByCount = (alpha, beta) => {
 };
 
 const getBarChartData = (state, ownProps) => {
-  if (!state.deviceReducer || !state.deviceReducer.devices || !state.kpiReducer || !state.kpiReducer.alarmsByRule) {
+  if (!state.deviceReducer || !state.deviceReducer.devices || !state.kpiReducer || !state.kpiReducer.alarmsByRule ||!state.ruleReducer.rulesAndActions) {
```

Comment:
Please split this conditions to separate lines.

Reply:
Okay. I am on it.